src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs

```diff
@@ -19,7 +19,15 @@ internal sealed class TdsParserStateObjectFactory
         //private static bool shouldUseLegacyNetorking;
         //public static bool UseManagedSNI { get; } = AppContext.TryGetSwitch(UseLegacyNetworkingOnWindows, out shouldUseLegacyNetorking) ? !shouldUseLegacyNetorking : true;
 
+#if DEBUG
+        private static Lazy<bool> shouldUseLegacyNetorking = new Lazy<bool>(
+            () => bool.TrueString.Equals(Environment.GetEnvironmentVariable("System.Data.SqlClient.UseLegacyNetworkingOnWindows"), StringComparison.InvariantCultureIgnoreCase)
+        );
+
+        public static bool UseManagedSNI => shouldUseLegacyNetorking.Value;
```

Comment:
Legacy is native SNI. IIUC, the environment variable, if set to true, causes Managed SNI to be used, which is non-legacy.

Reply:
Basically UseManagedSNI = True iff useLegacyNetworking is false.

Reply:
```
private static Lazy<bool> useManagedSNIOnWindows= new Lazy<bool>(
             () => bool.TrueString.Equals(Environment.GetEnvironmentVariable("System.Data.SqlClient.UseManagedSNIOnWindows"), StringComparison.InvariantCultureIgnoreCase)
);
public static bool UseManagedSNI => useManagedSNIOnWindows.Value;
```

Reply:
I think having two switches for the same thing with slightly different names would confuse things. I'll switch around the logic so that the switch is always `System.Data.SqlClient.UseLegacyNetworkingOnWindows` and that in debug mode it can be set by environment variable.

Considerations for the future and another PR; Is this switch public and documented and/or considered final? When I searched I could only find source hits in this repository. If it is public i'd like to think about re-instating the existing AppContext switch, if it isn't public i'd like to to consider if we can enable both AppContext and environment variable modes because it isn't always possible to recompile when you want to change a behaviour.

Reply:
I was not recommending two different switches. I was asking for a rename of the switch that you have created. 
This history is that we were going to deprecate Native Sni.dll on Windows and allow AppContext to switch to switch back to native sni.dll. However with the change you are making, you want managed SNI to be opt-in. Hence the recommended rename.

Reply:
I agree with renaming my addition. 
If we ever re-enable the diabled AppContext switch do we have to keep the old name or can I change that to use the new name as well?

Reply:
And what does SNI stand for? i assumed it was Sql Native Implementation, in which case ManagedSNI doesn't sound right.

Reply:
The AppContext (if added) will be added to make Managed Implementation as Opt-Out. I imagine it being a different name.

Reply:
Sql Network Interface.

Reply:
I might put that in a comment somewhere obvious for the next person 😁 
Changes made and pushed. manual native tests passed. Waiting on a release full build so I can try the managed ones.